lib/cosmos/dart/lib/dart_importer.rb

```diff
@@ -117,27 +122,41 @@ def import(filename, force)
 
       # No PacketLogEntry was found so create one from scratch
       unless ple
-        ple = PacketLogEntry.new
-        ple.target_id = target_id
-        ple.packet_id = packet_id
-        ple.time = packet.received_time
-        ple.packet_log_id = packet_log.id
-        ple.data_offset = data_offset
-        ple.meta_id = meta_id
-        ple.is_tlm = is_tlm
-        ple.ready = true
-        ple.save!
-        count += 1
-
         # SYSTEM META packets are special in that their meta_id is their own
         # PacketLogEntry ID from the database. All other packets have meta_id
         # values which point back to the last SYSTEM META PacketLogEntry ID.
         if target_name == 'SYSTEM'.freeze and packet_name == 'META'.freeze
+          if ple_data.length > 0
+            ActiveRecord::Base.connection.execute("INSERT INTO packet_log_entries (created_at, updated_at, target_id, packet_id, time, packet_log_id, data_offset, meta_id, is_tlm, ready) VALUES #{ple_data}")
+            ple_data.clear
+            ple_data_count = 0
+          end
+
+          ple = PacketLogEntry.new
+          ple.target_id = target_id
+          ple.packet_id = packet_id
+          ple.time = packet.received_time
+          ple.packet_log_id = packet_log.id
+          ple.data_offset = data_offset
+          ple.meta_id = meta_id
+          ple.is_tlm = is_tlm
+          ple.ready = true
+          ple.save!(validate: false)
+
           # Need to update meta_id for this and all subsequent packets
           meta_id = ple.id
           ple.meta_id = meta_id
-          ple.save!
+          ple.save!(validate: false)
+        else
+          if ple_data.length == 0
+            ple_data << "('#{entry_time}','#{entry_time}',#{target_id},#{packet_id},'#{packet.received_time.dup.utc.to_s(:db)}',#{packet_log.id},#{data_offset},#{meta_id},#{is_tlm},true)"
+          else
+            ple_data << ",('#{entry_time}','#{entry_time}',#{target_id},#{packet_id},'#{packet.received_time.dup.utc.to_s(:db)}',#{packet_log.id},#{data_offset},#{meta_id},#{is_tlm},true)"
+          end
```

Comment:
How about 152 by itself and then:
```
# Prepend a comma since we already have some ple_data items
ple_data.prepend(",") if ple_data.length != 0
```
Or I might even add the comma at the end every time and then remove the last one with ple_data[0..-2].